Modules/StudyProgramme/classes/model/class.ilStudyProgrammeType.php

```diff
@@ -570,14 +572,24 @@ public function processAndStoreIconFile(array $file_data) {
 		if (!count($file_data) || !$file_data['name']) {
 			return false;
 		}
-		if (!is_dir($this->getIconPath())) {
-			ilUtil::makeDirParents($this->getIconPath());
+		if (!$this->g_webdir->hasDir($this->getIconPath())) {
+			$this->g_webdir->createDir($this->getIconPath());
 		}
 		$filename = $this->getIcon() ? $this->getIcon() : $file_data['name'];
-		$return = ilUtil::moveUploadedFile($file_data['tmp_name'], $filename, $this->getIconPath(true), false);
-
-		// TODO Resize
-		return $return;
+		if($this->g_webdir->has($this->getIconPath(true)))
+		{
+			$this->g_webdir->delete($this->getIconPath(true));
+		}
+		try
+		{
+			$stream = $this->g_tmpdir->readStream(explode("/", $file_data["tmp_name"])[2]);
```

Comment:
This only works in very specific circumstances (upload-dir = temp-dir, temp-dir is on first level after /). We found a more robust solution with `fopen` (sadly) yesterday.

Reply:
`$this->tmp_dir` can be removed then.